modin/backends/pandas/parsers.py

```diff
@@ -149,18 +149,39 @@ def get_dtypes(cls, dtypes_ids):
 
         Returns
         -------
-        pandas.Series
-            pandas.Series where index is columns names and values are
-            columns dtypes.
+        frame_dtypes : pandas.Series or dtype
+            Resulting dtype or pandas.Series where column names are used as
+            index and types of columns are used as values for full resulting
+            frame.
         """
+        # each element in `partitions_dtypes` is a Series, where column names are
+        # used as index and types of columns for different partitions are used as values
         partitions_dtypes = cls.materialize(dtypes_ids)
         if all([len(dtype) == 0 for dtype in partitions_dtypes]):
             return None
-        return (
-            pandas.concat(partitions_dtypes, axis=1)
-            .apply(lambda row: find_common_type_cat(row.values), axis=1)
-            .squeeze(axis=0)
-        )
+
+        combined_part_dtypes = pandas.concat(partitions_dtypes, axis=1)
+        frame_dtypes = combined_part_dtypes.iloc[:, 0]
+
+        for part_dtype_col in combined_part_dtypes.columns:
```

Comment:
This for loop will still be costly for the common case where all are equal. I prefer to avoid the for loop with something like:

```python
if combined_part_dtypes.eq(combined_part_dtypes.iloc[:, 0], axis=1).all(axis=None):
    pass
```

This does the check in a non-looping way. The `eq` will do a column-wise comparison and `all(axis=None)` will check for all values being `True`.

Reply:
Ok, `combined_part_dtypes.eq` method was used.